In [2]:
#Importing libraries

import pandas as pd
import  matplotlib.pyplot as plt
import os

In [3]:
#All yearly data combined

dir = "Sales Data/"
files = []
df = pd.DataFrame

for file in os.listdir(dir):
    read_df = pd.read_csv(dir + file)
    files.append(read_df)

df = pd.concat(files)

df.dropna(inplace=True)
df.drop_duplicates(keep= False, inplace=True)
df.head(10)

Order ID                     Product Quantity Ordered Price Each  \
0    176558        USB-C Charging Cable                2      11.95   
2    176559  Bose SoundSport Headphones                1      99.99   
3    176560                Google Phone                1        600   
4    176560            Wired Headphones                1      11.99   
5    176561            Wired Headphones                1      11.99   
6    176562        USB-C Charging Cable                1      11.95   
7    176563  Bose SoundSport Headphones                1      99.99   
8    176564        USB-C Charging Cable                1      11.95   
9    176565          Macbook Pro Laptop                1       1700   
10   176566            Wired Headphones                1      11.99   

        Order Date                        Purchase Address  
0   04/19/19 08:46            917 1st St, Dallas, TX 75001  
2   04/07/19 22:30       682 Chestnut St, Boston, MA 02215  
3   04/12/19 14:38    669 Spruce St, Los Angeles, CA 90001  
4   04/12/19 14:38    669 Spruce St, Los Angeles, CA 90001  
5   04/30/19 09:27       333 8th St, Los Angeles, CA 90001  
6   04/29/19 13:03  381 Wilson St, San Francisco, CA 94016  
7   04/02/19 07:46        668 Center St, Seattle, WA 98101  
8   04/12/19 10:58         790 Ridge St, Atlanta, GA 30301  
9   04/24/19 10:38  915 Willow St, San Francisco, CA 94016  
10  04/08/19 14:05             83 7th St, Boston, MA 02215

# What was the best month for sales? How much was earned that month?

In [4]:
#Add Month amd Year column to filter data by dates
df["Order Date"] = pd.to_datetime(df["Order Date"], errors="coerce")

df["Month"] = df["Order Date"].dt.month.astype(int)
df["Year"] = df["Order Date"].dt.year.astype(int)


In [5]:
#Add a column to get order total per Order ID

# df["Order Total"] = df["Quantity Ordered"] * float(df["Price Each"]
# df.head(10)

#Error because types are objects not numbers
df.dtypes

Order ID                    object
Product                     object
Quantity Ordered            object
Price Each                  object
Order Date          datetime64[ns]
Purchase Address            object
Month                        int32
Year                         int32
dtype: object

In [6]:
df.dropna(inplace= True)
df["Quantity Ordered"] = pd.to_numeric(df["Quantity Ordered"])
df["Price Each"] = pd.to_numeric(df["Price Each"])
df.dtypes


Order ID                    object
Product                     object
Quantity Ordered             int64
Price Each                 float64
Order Date          datetime64[ns]
Purchase Address            object
Month                        int32
Year                         int32
dtype: object

In [7]:
df["Order Total"] = df["Quantity Ordered"].astype(float) * df["Price Each"]
df.head(10)

Order ID                     Product  Quantity Ordered  Price Each  \
0    176558        USB-C Charging Cable                 2       11.95   
2    176559  Bose SoundSport Headphones                 1       99.99   
3    176560                Google Phone                 1      600.00   
4    176560            Wired Headphones                 1       11.99   
5    176561            Wired Headphones                 1       11.99   
6    176562        USB-C Charging Cable                 1       11.95   
7    176563  Bose SoundSport Headphones                 1       99.99   
8    176564        USB-C Charging Cable                 1       11.95   
9    176565          Macbook Pro Laptop                 1     1700.00   
10   176566            Wired Headphones                 1       11.99   

            Order Date                        Purchase Address  Month  Year  \
0  2019-04-19 08:46:00            917 1st St, Dallas, TX 75001      4  2019   
2  2019-04-07 22:30:00       682 Chestnut St, Boston, MA 02215      4  2019   
3  2019-04-12 14:38:00    669 Spruce St, Los Angeles, CA 90001      4  2019   
4  2019-04-12 14:38:00    669 Spruce St, Los Angeles, CA 90001      4  2019   
5  2019-04-30 09:27:00       333 8th St, Los Angeles, CA 90001      4  2019   
6  2019-04-29 13:03:00  381 Wilson St, San Francisco, CA 94016      4  2019   
7  2019-04-02 07:46:00        668 Center St, Seattle, WA 98101      4  2019   
8  2019-04-12 10:58:00         790 Ridge St, Atlanta, GA 30301      4  2019   
9  2019-04-24 10:38:00  915 Willow St, San Francisco, CA 94016      4  2019   
10 2019-04-08 14:05:00             83 7th St, Boston, MA 02215      4  2019   

    Order Total  
0         23.90  
2         99.99  
3        600.00  
4         11.99  
5         11.99  
6         11.95  
7         99.99  
8         11.95  
9       1700.00  
10        11.99

In [8]:
monthly_totals = df.groupby("Month").sum()
monthly_totals.drop(["Price Each", "Year"], axis= 1, inplace=True)
monthly_totals

Quantity Ordered  Order Total
Month                               
1                 10883   1820569.59
2                 13413   2198133.74
3                 16953   2802846.32
4                 20514   3387765.72
5                 18639   3148625.71
6                 15215   2574758.04
7                 16036   2645146.88
8                 13410   2237698.86
9                 13073   2091371.25
10                22635   3732828.84
11                19740   3196146.90
12                28034   4603148.06

In [9]:
monthly_totals[monthly_totals["Order Total"] == monthly_totals["Order Total"].max()]

Quantity Ordered  Order Total
Month                               
12                28034   4603148.06

### December was the highest grossing month of the year with £4,603,148.06 in revenue
---

# What time should we display advertisements to maximise the likelihood of customers buying a product?

In [13]:
df["Hour"] = df["Order Date"].dt.hour
hourly_sales = df.groupby("Hour").sum()
hourly_sales.sort_values("Quantity Ordered", ascending=False).head(1).drop(["Price Each", "Month", "Year"], axis=1)

Quantity Ordered  Order Total
Hour                               
19               14432   2411003.74

In [12]:
hourly_sales.sort_values("Order Total", ascending= False).head(1).drop(["Price Each", "Month", "Year"], axis=1)

Quantity Ordered  Order Total
Hour                               
19               14432   2411003.74

### The timeslot where most orders are place fall between 7pm -8pm. Advertising should be scheduled to take advantage of this time slot.
---

# What products are most often sold together?